acra-censor/common/logging_logic.go

```diff
@@ -0,0 +1,285 @@
+package common
+
+import (
+	"bytes"
+	"encoding/json"
+	"github.com/cossacklabs/acra/logging"
+	"github.com/cossacklabs/acra/sqlparser"
+	log "github.com/sirupsen/logrus"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+)
+
+// DefaultSerializationTimeout defines a default ticker' timeout
+const DefaultSerializationTimeout = time.Second
+
+// QueryInfo defines format of exporting query into file
+type QueryInfo struct {
+	RawQuery    string `json:"raw_query"`
+	IsForbidden bool   `json:"_blacklisted_by_web_config"`
+}
+
+// QueryWriter is a mechanism that provides dumping stored queries into file in background
+type QueryWriter struct {
+	filePath             string
+	Queries              []*QueryInfo
+	BufferedQueries      []*QueryInfo
+	signalBackgroundExit chan bool
+	serializationTimeout time.Duration
+	serializationTicker  *time.Ticker
+	logger               *log.Entry
+}
+
+// NewQueryWriter creates QueryWriter instance
+func NewQueryWriter(filePath string) (*QueryWriter, error) {
+	// open or create file, APPEND MODE
+	openedFile, err := os.OpenFile(filePath, os.O_APPEND|os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		log.WithField("internal_object", "querywriter").WithError(ErrCantReadQueriesFromFileError).WithField(logging.FieldKeyEventCode, logging.EventCodeErrorCensorIOError).Errorln("Can't create QueryWriter instance")
+		return nil, err
+	}
+
+	// signals
+	signalShutdown := make(chan os.Signal, 2)
+	signal.Notify(signalShutdown, os.Interrupt, syscall.SIGTERM)
+	signalBackgroundExit := make(chan bool)
+
+	// create handler
+	handler := &QueryWriter{}
```

Comment:
I think we should assign `openedFile` to the related writer which use it to not forget to close it on Release/Close. resource owner responsible for releasing resources. plus it allow to avoid extra OpenFile calls because file already opened